### PR TITLE
Some connection namespace moves

### PR DESCRIPTION
**Pull request by elmarco**

### CODE REVIEW

File: book/src/connection.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ let (client_conn, server_conn) = futures_util::try_join!(
 [`futures::stream::Stream`]: https://docs.rs/futures/latest/futures/stream/trait.Stream.html
 [`futures::sink::Sink`]: https://docs.rs/futures/latest/futures/sink/trait.Sink.html
 [`MessageStream`]: https://docs.rs/zbus/3/zbus/struct.MessageStream.html
-[`ConnectionBuilder::address`]: https://docs.rs/zbus/3/zbus/struct.ConnectionBuilder.html#method.address 
+[`Builder::address`]: https://docs.rs/zbus/3/zbus/struct.ConnectionBuilder.html#method.address
```
Comment (zeenix) [inline book/src/connection.md:76]: The links will also need to be updated (you need to predict it based on a local build). Was this done for the other module changes? :thinking:

File: zbus/src/object_server/mod.rs [hunk 2]
```diff
@@ -18,10 +18,11 @@ use zvariant::{ObjectPath, OwnedObjectPath, OwnedValue, Signature, Type, Value};
 
 use crate::{
     async_lock::{RwLock, RwLockReadGuard, RwLockWriteGuard},
+    connection::WeakConnection,
```
Comment (zeenix) [inline zbus/src/object_server/mod.rs:21]: maybe this should just be `connection::Weak`, similar to `std::rc::Weak` and `std::sync::Weak`?

**Review by zeenix — CHANGES_REQUESTED** [review]
Nice job otherwise.

File: book/src/connection.md (hunk 1 above)
Comment (elmarco) [reply book/src/connection.md:76]: I am bad at predicting it though.. is there any way to update it or check it semi-automatically? yes, we will probably need to update all the links for v4 anyway..

File: zbus/src/object_server/mod.rs (hunk 2 above)
Comment (elmarco) [reply zbus/src/object_server/mod.rs:21]: yes, I thought about it and forgot, let's do it

Comment (elmarco) [reply zbus/src/object_server/mod.rs:21]: nah that conflicts with other Weak types.. it's internal anyway, not a big deal.

File: book/src/connection.md (hunk 1 above)
Comment (zeenix) [reply book/src/connection.md:76]: `cargo doc --open` and then find the link?

Comment (elmarco) [reply book/src/connection.md]: done

**Review by zeenix — APPROVED** [review]

**Pull request merged**